### PR TITLE
add data type DataValue

### PR DESCRIPTION
**Pull request by zemirco**
Here is the next data type `DataValue`.

### CODE REVIEW

**Comment by zemirco** [comment]
this is work in progress. i want to add another test with more fields actually used.

**Comment by wmnsk** [comment]
Cool!

Could you add constructor `NewDataValue()`?
I understand constructor may not be so useful for this kind of “conditionally formatted” type, but it also helps users handle any kind of type in the same way.

**Comment by zemirco** [comment]
OK I'm done. PR is ready for review.

File: datatypes/data-value.go [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+	Value             *Variant
+	Status            uint32
+	SourceTimestamp   time.Time
+	SourcePicoSeconds uint16
+	ServerTimestamp   time.Time
+	ServerPicoSeconds uint16
+}
+
+// NewDataValue creates a new DataValue.
+func NewDataValue(hasValue, hasSrcTs, hasSrcPs, hasSvrTs, hasServerPc bool, v *Variant, status uint32, srcTs time.Time, srcPs uint16, svrTs time.Time, svrPs uint16) *DataValue {
```
Comment (wmnsk) [inline datatypes/data-value.go]: Can you remane `hasServerPc` as `hasSvrPs` for consistency?

Comment (zemirco) [reply datatypes/data-value.go]: ah sorry about that

**Comment by wmnsk** [comment]
Thanks. Now it looks good to me. Merging.

**Pull request merged**